metricbeat/module/prometheus/_meta/run.sh

```diff
@@ -1,12 +1,20 @@
 #!/bin/sh
 
-# this works only on Mac envs
-HOST_DOMAIN="host.docker.internal"
-ping -q -c1 $HOST_DOMAIN > /dev/null 2>&1
-if [ $? -ne 0 ]; then
-  # this works only on Linux envs
-  HOST_DOMAIN="0.0.0.0"
-fi
+
+for i in 1 2 3 4 5;
+do
+  a=`nslookup host.docker.internal | grep "** server can't find " | wc -l`;
+  if [ $a -gt 0 ]; then
+    # this works only on Linux envs
+    HOST_DOMAIN="0.0.0.0"
+  else
+    # this works only on Mac envs
+    HOST_DOMAIN="host.docker.internal"
+    break
```

Comment:
Why only breaking in this case?

Reply:
We want to keep trying to reach `host.docker.internal` otherwise to fallback to `0.0.0.0`.